core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -879,8 +879,19 @@ private boolean checkConsistencyInternal(Inode inode, AlluxioURI path)
       return ufs.isDirectory(ufsPath);
     } else {
       InodeFile file = (InodeFile) inode;
-      return ufs.isFile(ufsPath)
-          && ufs.getFileStatus(ufsPath).getContentLength() == file.getLength();
+      if (ufs.isFile(ufsPath)) {
+        UfsFileStatus ufsFileStatus = ufs.getFileStatus(ufsPath);
+        if (ufs.getFileStatus(ufsPath).getContentLength() != file.getLength()) {
+          return false;
+        } else if (hasAnyBlockCached(mBlockMaster.getBlockInfoList(file.getBlockIds()))
+            && (ufsFileStatus.getLastModifiedTime() > file.getLastModificationTimeMs())) {
```

Comment:
@aaudiber 

This is really a big issue. I think we can use UFS clock to calculate inconsistent.

So, i should add a field to `INode` named like 'mUFSLastModificationTimeMs', when a INode created, assigned ufs LastModificationTime to this field. When we check the consistency, we compare this feild and UFS file LastModificationTime. 

Has any other better idea?